### PR TITLE
PageStorage: Fix exception caused by empty page

### PR DESCRIPTION
**Pull request by JaySon-Huang**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tiflash/issues/8018

Problem Summary:
In the previous implement, we assume that the page_data is non-empty. When there are page with `size==0`, we may run into exception when restoring PageStorage entries from disk

As the issue descripe, because all error page_id with `size==0` are ends with '0x07', I think it is comes from `dirty_mark` in raft_log_engine

https://github.com/tikv/tikv/blob/6560d758f9143dc5125b0c5c3b0eaadbfecffa3c/components/raft_log_engine/src/engine.rs#L486-L496

### What is changed and how it works?

Ignore the error when we are restoring entries from disk when `size==0`

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/Page/V3/Blob/BlobStat.cpp [hunk 1]
```diff
@@ -329,20 +345,16 @@ size_t BlobStats::BlobStat::removePosFromStat(
...
-            fmt::format(
-                "Restore position from BlobStat failed, the space/subspace is already being used [offset={}] "
-                "[buf_size={}] [blob_id={}]",
-                offset,
-                buf_size,
-                id),
-            ErrorCodes::LOGICAL_ERROR);
+        return std::make_tuple(success, smap->toDebugString());
+        String msg = (buf_size == 0) ? "" : smap->toDebugString();
+        return std::make_tuple(success, msg);
```
[the remark this answers is not in the document]
Comment (JaySon-Huang) [reply dbms/src/Storages/Page/V3/Blob/BlobStat.cpp:354]: Good catch! I've fix it

**Review by lidezhu — APPROVED** [review]
LGTM

**Pull request merged**